optuna/pruners/threshold.py

```diff
@@ -0,0 +1,59 @@
+from optuna.pruners import BasePruner
+
+from optuna.type_checking import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+    from optuna.structs import FrozenTrial
+    from optuna.study import Study
+
+
+class ThresholdPruner(BasePruner):
+    """Pruner to detect abnormal metrics of the trials.
+
+    Prune if a metric exceeds upper bound threshold or
+    falls behind lower bound threshold which users specify.
+
+    Example:
+
+        .. code::
```

Comment:
We could address this in a follow up but would you find it possible to rewrite this example as a doctest similar to https://github.com/optuna/optuna/pull/958/files?

Reply:
I tried to create a doctest in 868a857. (but `make doctest` seems to ignore this...)

Reply:
Ah, you have to add the `ThresholdPruner` to https://github.com/optuna/optuna/blob/master/docs/source/reference/pruners.rst.

Reply:
Thank you, I memorized. :memo: